src/core/server/elasticsearch/elasticsearch_config.test.ts

```diff
@@ -378,6 +378,50 @@ describe('deprecations', () => {
     const { messages } = applyElasticsearchDeprecations({ ssl: { key: '', certificate: '' } });
     expect(messages).toEqual([]);
   });
+
+  it('logs a warning if requestHeadersWhitelist contains the system index header', () => {
+    // Array format
+    const { messages: messages1 } = applyElasticsearchDeprecations({
+      requestHeadersWhitelist: ['x-elastic-product-origin'],
+    });
+    expect(messages1).toMatchInlineSnapshot(`
+      Array [
+        "Setting [elasticsearch.requestHeadersWhitelist] contains \\"x-elastic-product-origin\\", but in a future version, direct access to system indices will be prevented by default",
```

Comment:
These warnings closely mirror the language used in the Elasticsearch deprecation warning header:

> "this request accesses system indices: [.kibana_8.0.0_001], but in a future major version, direct access to system indices will be prevented by default"